packages/hooks/src/index.js

```diff
@@ -0,0 +1,333 @@
+ * @param {string}  action The name of the action to check for.
+ *
+ * @return {[type]}      [description]
+ */
+function createCurrentHookByType( type ) {
+	return function( action ) {
+
+		// If the action was not passed, check for any current hook.
+		if ( 'undefined' === typeof action ) {
+			return false;
+		}
+
+		// Return the current hook.
+		return HOOKS[ type ] && HOOKS[ type ].current ?
+			HOOKS[ type ].current :
+			false;
+	};
+}
+
+
+
+/**
+ * Checks to see if an action is currently being executed.
+ *
+ * @param  {string} type   Type of hooks to check, one of 'action' or 'filter'.
+ * @param {string}  action The name of the action to check for, if omitted will check for any action being performed.
+ *
+ * @return {[type]}      [description]
+ */
+function createDoingHookByType( type ) {
+	return function( action ) {
+
+		// If the action was not passed, check for any current hook.
+		if ( 'undefined' === typeof action ) {
+			return 'undefined' !== typeof HOOKS[ type ].current;
+		}
+
+		// Return the current hook.
+		return HOOKS[ type ] && HOOKS[ type ].current ?
+			action === HOOKS[ type ].current :
+			false;
+	};
+}
+
+/**
+ * Retrieve the number of times an action is fired.
+ *
+ * @param  {string} type   Type for which hooks to check, one of 'action' or 'filter'.
+ * @param {string}  action The action to check.
+ *
+ * @return {[type]}      [description]
+ */
+function createDidHookByType( type ) {
+	return function( action ) {
+		return HOOKS[ type ] && HOOKS[ type ][ action ] && HOOKS[ type ][ action ].runs ?
+			HOOKS[ type ][ action ].runs :
+			0;
+	};
+}
+
+/**
+ * Check to see if an action is registered for a hook.
+ *
+ * @param  {string} type   Type for which hooks to check, one of 'action' or 'filter'.
+ * @param {string}  action  The action to check.
+ *
+ * @return {bool}      Whether an action has been registered for a hook.
+ */
+function createHasHookByType( type ) {
+	return function( action ) {
+		return HOOKS[ type ] && HOOKS[ type ][ action ] ?
+			!! HOOKS[ type ][ action ] :
+			false;
+	};
+}
+
+/**
+ * Remove all the actions registered to a hook,
+ */
+function createRemoveAllByType( type ) {
+	return function( action, type ) {
+
+	};
+}
+
+// Remove functions.
+export removeFilter = createRemoveHookByType( 'filters' );
+export removeAction = createRemoveHookByType( 'actions' );
+
+
+// Do action/apply filter functions.
+export doAction =     createRunHookByType( 'actions', runDoAction );
+export applyFilters = createRunHookByType( 'filters', runApplyFilters );
+
+// Add functions.
+export addAction = createAddHookByType( 'actions' );
+export addFilter = createAddHookByType( 'filters' );
+
+// Doing functions.
+export doingAction = createDoingHookByType( 'actions' ), /* True for actions until;next action fired. */
+export doingFilter = createDoingHookByType( 'filters' ), /* True for filters while;filter is being applied. */
```

Comment:
Why the semicolons in the middle of these comments?  Should be `//` inline comments as well